lib/mongoid/extensions/hash.rb

```diff
@@ -166,6 +166,24 @@ def to_criteria
 
       private
 
+      # Get the value for the provided operator, klass, key and value.
+      #
+      # This is necessary for special cases like $rename, $addToSet and $push.
+      #
+      # @param [ String ] operator The operator.
+      # @param [ Class ] klass The model class.
+      # @param [ String | Symbol ] key The field key.
+      # @param [ Object ] value The original value.
+      #
+      # @return [ Object ] Value prepared for the provided operator.
+      def value_for(operator, klass, key, value)
```

Comment:
I propose any new monkey patch method (private or otherwise) should have `__` prefix.